### PR TITLE
Refactor for parser initialization and data preprocessing

### PR DESCRIPTION
**Pull request by MAYAS3**

### CODE REVIEW

File: deepparse/converter/data_processor_factory.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            processor = DataProcessor(
+                vectorizer, padder.pad_subword_embeddings_sequences, padder.pad_subword_embeddings_batch, tags_converter
+            )
+
+        else:
+            processor = DataProcessor(
+                vectorizer, padder.pad_word_embeddings_sequences, padder.pad_word_embeddings_batch, tags_converter
+            )
+
+        return processor
```
Comment (davebulaval) [inline deepparse/converter/data_processor_factory.py:31]: Maybe use and elif(FastTextVectoriser) ... else RaiseError(...)?

File: deepparse/parser/address_parser.py [hunk 2]
```diff
@@ -1044,61 +1034,33 @@ def _model_factory(
...
+            model_type=self.model_type,
+            cache_dir=cache_dir,
+            device=self.device,
+            output_size=prediction_layer_len,
+            attention_mechanism=attention_mechanism,
+            path_to_retrained_model=path_to_retrained_model,
+            offline=offline,
+            verbose=verbose,
+            **seq2seq_kwargs,
+        )
```
Comment (davebulaval) [inline deepparse/parser/address_parser.py:1048]:
```suggestion
        self.model = ModelFactory().create(
            model_type=self.model_type,
            cache_dir=cache_dir,
            device=self.device,
            output_size=prediction_layer_len,
            attention_mechanism=attention_mechanism,
            path_to_retrained_model=path_to_retrained_model,
            offline=offline,
            verbose=verbose,
            **seq2seq_kwargs,
        )
```

Comment (davebulaval) [reply deepparse/parser/address_parser.py:1048]: I don't think we need to keep the factory as a variable object.

**Review by davebulaval — DISMISSED** [review]
Good refactor the factories; it was about time.

File: deepparse/parser/address_parser.py (hunk 2 above)
Comment (davebulaval) [reply deepparse/parser/address_parser.py:1048]: Also, let's change the method name. Maybe something like _model_setup or else.

**Comment by davebulaval** [comment]
GPU tests pass. For me it is a go.

File: deepparse/converter/data_processor_factory.py [hunk 3]
```diff
@@ -23,9 +23,16 @@ def create(self, vectorizer: Vectorizer, padder: DataPadder, tags_converter: Tag
...
 
-        else:
+        elif isinstance(vectorizer, (FastTextVectorizer, MagnitudeVectorizer)):
             processor = DataProcessor(
                 vectorizer, padder.pad_word_embeddings_sequences, padder.pad_word_embeddings_batch, tags_converter
             )
+        else:
+            raise NotImplementedError(
+                """
+            There's no data processor corresponding to the  provided vectorizer.
```
Comment (davebulaval) [inline deepparse/converter/data_processor_factory.py:33]:
```suggestion
            There's no data processor corresponding to the provided vectorizer.
```

**Review by davebulaval — APPROVED** [review]

File: deepparse/converter/data_processor_factory.py (hunk 1 above)
Comment (MAYAS3) [reply deepparse/converter/data_processor_factory.py]: Done!

File: deepparse/parser/address_parser.py (hunk 2 above)
Comment (MAYAS3) [reply deepparse/parser/address_parser.py]: Done!

**Pull request merged**